src/wazuh_modules/syscollector/sysInfoWin.cpp

```diff
@@ -0,0 +1,170 @@
+{
+    BYTE Type;
+    BYTE FormattedAreaLength;
+    WORD Handle;
+    BYTE Manufacturer;
+    BYTE Product;
+    BYTE Version;
+    BYTE SerialNumber;
+} SMBIOSBasboardInfoStructure;
+
+static bool isVistaOrLater()
+{
+    static const bool ret
+    {
+        IsWindowsVistaOrGreater()
+    };
+    return ret;
+}
+
+/* Reference: https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_2.6.0.pdf */
+static std::string parseRawSmbios(const BYTE* rawData, const DWORD rawDataSize)
+{
+    std::string serialNumber;
+    DWORD offset{0};
+    while (offset < rawDataSize && serialNumber.empty())
+    {
+        SMBIOSStructureHeader header{};
+        memcpy(&header, rawData + offset, sizeof(SMBIOSStructureHeader));
+        if (header.Type == BASEBOARD_INFORMATION_TYPE)
+        {
+            SMBIOSBasboardInfoStructure info{};
+            memcpy(&info, rawData + offset, sizeof(SMBIOSBasboardInfoStructure));
+            offset += info.FormattedAreaLength;
+            for (BYTE i = 1; i < info.SerialNumber; ++i)
+            {
+                const char* tmp{reinterpret_cast<const char*>(rawData + offset)};
+                const auto len{ strlen(tmp) };
+                offset += len + sizeof(char);
+            }
+            serialNumber = reinterpret_cast<const char*>(rawData + offset);
+        }
+        else
+        {
+            offset += header.FormattedAreaLength;
+            bool end{false};
+            while(!end)
+            {
+                const char* tmp{reinterpret_cast<const char*>(rawData + offset)};
+                const auto len{strlen(tmp)};
+                offset += len + sizeof(char);
+                end = !len;
+            }
+        }
+    }
+    return serialNumber;
+}
+
+std::string SysInfo::getSerialNumber()
+{
+    std::string ret;
+    if (isVistaOrLater())
+    {
+        const auto rawData{Utils::exec("wmic baseboard get SerialNumber")};
+        const auto pos{rawData.find("\r\n")};
+        if (pos != std::string::npos)
+        {
+            ret = Utils::trim(rawData.substr(pos), " \t\r\n");
+        }
+        else
+        {
+            ret = "unknown";
+        }
+    }
+    else
+    {
+        const auto size {GetSystemFirmwareTable('RSMB', 0, nullptr, 0)};
+        if (size)
+        {
+            std::unique_ptr<unsigned char> buff{new unsigned char[size]};
+            if (buff)
+            {
+                /* Get raw SMBIOS firmware table */
+                if (GetSystemFirmwareTable('RSMB', 0, buff.get(), size) == size)
+                {
+                    PRawSMBIOSData smbios{reinterpret_cast<PRawSMBIOSData>(buff.get())};
+                    /* Parse SMBIOS structures */
+                    ret = parseRawSmbios(smbios->SMBIOSTableData, size);
+                }
+            }
+        }
+    }
+    return ret;
+}
+
+std::string SysInfo::getCpuName()
+{
+    Utils::Registry reg(HKEY_LOCAL_MACHINE, "HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0");
```

Comment:
Maybe you can create a constexpr auto with this value and use it here (and in the line 142) and in the UTs.